src/MongoDB.Driver.Core/Core/Operations/UpdateRequest.cs

```diff
@@ -42,15 +42,15 @@ public sealed class UpdateRequest : WriteRequest
         /// <param name="updateType">The update type.</param>
         /// <param name="filter">The filter.</param>
         /// <param name="update">The update.</param>
-        public UpdateRequest(UpdateType updateType, BsonDocument filter, BsonDocument update)
+        public UpdateRequest(UpdateType updateType, BsonDocument filter, BsonValue update)
             : base(WriteRequestType.Update)
         {
             _updateType = updateType;
             _filter = Ensure.IsNotNull(filter, nameof(filter));
             _update = Ensure.IsNotNull(update, nameof(update));
-            if (updateType == UpdateType.Update && _update.ElementCount == 0)
+            if (updateType == UpdateType.Update)
             {
-                throw new ArgumentException("Updates must have at least 1 update operator.", nameof(update));
+                ValidateUpdateValue(_update);
```

Comment:
If you rename ValidateUpdateValue and refactor appropriately you can replace lines 50-54 with:

    _update = EnsureUpdateIsValid(update, updateType);